_ml-commons-plugin/remote-models/batch-ingestion.md

```diff
@@ -0,0 +1,250 @@
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "endpoint": "api.openai.com",
+        "model": "gpt-3.5-turbo",
+        "input_docs_processed_step_size": 100
+    },
+    "credential": {
+        "openAI_key": "..."
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://${parameters.endpoint}/v1/chat/completions",
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages} }"
+        }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+The `parameters.input_docs_processed_step_size`parameter is used to set the maximum batch size for documents sent to a remote server. You can set this parameter to the maximum batch size the remote server can support or a smaller number for optimal performance.
+
+The response contains the connector ID for the newly created connector:
+
+```json
+{
+  "connector_id": "a1eMb4kBJ1eYAeTMAljY"
+}
+```
+
+## Step 3: Register an externally hosted model
+
+To register an externally hosted model to the model group created in step 1, provide the model group ID from step 1 and the connector ID from step 2 in the following request. You must specify the `function_name` as `remote`:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "openAI-gpt-3.5-turbo",
+    "function_name": "remote",
+    "model_group_id": "wlcnb4kBJ1eYAeTMHlV6",
+    "description": "test model",
+    "connector_id": "a1eMb4kBJ1eYAeTMAljY"
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch returns the task ID of the register operation:
+
+```json
+{
+  "task_id": "cVeMb4kBJ1eYAeTMFFgj",
+  "status": "CREATED"
+}
+```
+
+To check the status of the operation, provide the task ID to the [Tasks API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/tasks-apis/get-task/):
+
+```json
+GET /_plugins/_ml/tasks/cVeMb4kBJ1eYAeTMFFgj
+```
+{% include copy-curl.html %}
+
+When the operation is complete, the state changes to `COMPLETED`:
+
+```json
+{
+  "model_id": "cleMb4kBJ1eYAeTMFFg4",
+  "task_type": "REGISTER_MODEL",
+  "function_name": "REMOTE",
+  "state": "COMPLETED",
+  "worker_node": [
+    "XPcXLV7RQoi5m8NI_jEOVQ"
+  ],
+  "create_time": 1689793598499,
+  "last_update_time": 1689793598530,
+  "is_async": false
+}
+```
+
+Take note of the returned `model_id` because you’ll need it to deploy the model.
+
+## Step 4: Deploy the model
+
+Starting with OpenSearch version 2.13, externally hosted models are deployed automatically by default when you send a Predict API request for the first time. To disable automatic deployment for an externally hosted model, set `plugins.ml_commons.model_auto_deploy.enable` to `false`:
```

Comment:
Although technically accurate, "automatically by default" reads as redundant. Can we use just one of them?